code/handlers/trackservers.q

```diff
@@ -243,96 +243,109 @@ procupdate:{[procs] addprocs[procs;exec distinct proctype from procs;0b];}
                         procs,:([]host:`;port:0Ni;proctype:`discovery;procname:`;hpup:.servers.DISCOVERY)]];
-	// Remove any processes that have an active connection
-	connectedprocs: select procname, proctype, hpup from SERVERS;
-	procs: delete from procs where ([] procname; proctype; hpup) in connectedprocs;
-	nontorqprocs: delete from nontorqprocesstab where ([] procname; proctype; hpup) in connectedprocs;
-	// if there aren't any processes left to connect to, then escape
-	if[not any count each (procs;nontorqprocs); .lg.o[`conn;"No new processes to connect to.  Escaping..."];:()];
-	if[CONNECTIONSFROMDISCOVERY or DISCOVERYREGISTER;
-		register[procs;`discovery;0b];
-		retrydiscovery[]];
-	if[not CONNECTIONSFROMDISCOVERY; register[procs;;0b] each $[CONNECTIONS~`ALL;exec distinct proctype from procs;CONNECTIONS]];
-	if[TRACKNONTORQPROCESS;register[nontorqprocs;;0b] each  $[CONNECTIONS~`ALL;exec distinct proctype from nontorqprocs;CONNECTIONS]];
-	// try and open dead connections
-	retry[]}
+    // Remove any processes that have an active connection
+    connectedprocs: select procname, proctype, hpup from SERVERS;
+    procs: delete from procs where ([] procname; proctype; hpup) in connectedprocs;
+    nontorqprocs: delete from nontorqprocesstab where ([] procname; proctype; hpup) in connectedprocs;
+    // if there aren't any processes left to connect to, then escape
+    if[not any count each (procs;nontorqprocs); .lg.o[`conn;"No new processes to connect to.  Escaping..."];:()];
+    if[CONNECTIONSFROMDISCOVERY or DISCOVERYREGISTER;
+        register[procs;`discovery;0b];
+        retrydiscovery[]];
+    if[not CONNECTIONSFROMDISCOVERY; register[procs;;0b] each $[CONNECTIONS~`ALL;exec distinct proctype from procs;CONNECTIONS]];
+    if[TRACKNONTORQPROCESS;register[nontorqprocs;;0b] each  $[CONNECTIONS~`ALL;exec distinct proctype from nontorqprocs;CONNECTIONS]];
+    // try and open dead connections
+    retry[]}
 
 
 pc:{[result;W] update w:0Ni,endp:.proc.cp[] from`.servers.SERVERS where w=W;cleanup[];result}
 
 if[enabled;
-	.z.pc:{.servers.pc[x y;y]}.z.pc;
-	if[DISCOVERYRETRY > 0; .timer.repeat[.proc.cp[];0Wp;DISCOVERYRETRY;(`.servers.retrydiscovery;`);"Attempt reconnections to the discovery service"]];
-	if[RETRY > 0; .timer.repeat[.proc.cp[];0Wp;RETRY;(`.servers.retry;`);"Attempt reconnections to closed server handles"]]];
+    .z.pc:{.servers.pc[x y;y]}.z.pc;
+    if[DISCOVERYRETRY > 0; .timer.repeat[.proc.cp[];0Wp;DISCOVERYRETRY;(`.servers.retrydiscovery;`);"Attempt reconnections to the discovery service"]];
+    if[RETRY > 0; .timer.repeat[.proc.cp[];0Wp;RETRY;(`.servers.retry;`);"Attempt reconnections to closed server handles"]]];
+
+// Check if required processes all connected
+reqprocsnotconn:{[requiredprocs] 
+    not all requiredprocs in exec proctype from .servers.SERVERS where w in key .z.W
+  }
+
+// Block process until all required processes are connected
+startupdependent:{[requiredprocs;timeintv]
```

Comment:
Could you move this and previous function to be defined just after start up (before pc)